scripts/test-runner.js

```diff
@@ -0,0 +1,238 @@
+                });
+                rcProcess.on('close', () => {
+                    fs.closeSync(outFD);
+                    fs.closeSync(errFD);
+                });
+            } catch (err) {
+                throw err;
+            }
+        }
+    } else {
+        if (background) {
+            rcProcess = spawn(
+                  `nohup java -Dhazelcast.enterprise.license.key=${HAZELCAST_ENTERPRISE_KEY} `
+                + `-cp ${CLASSPATH} com.hazelcast.remotecontroller.Main --use-simple-server > rc_stdout.log `
+                + '2> rc_stderr.log &', [], {
+                stdio: 'ignore',
+                shell: true
+            });
+        } else {
+            try {
+                const outFD = fs.openSync('rc_stdout.log', 'w');
+                const errFD = fs.openSync('rc_stderr.log', 'w');
+                rcProcess = spawn('java', [
+                    `-Dhazelcast.enterprise.license.key=${HAZELCAST_ENTERPRISE_KEY}`,
+                    '-cp',
+                    CLASSPATH,
+                    'com.hazelcast.remotecontroller.Main',
+                    '--use-simple-server'
+                ], {
+                    stdio: [
+                        'ignore',
+                        outFD,
+                        errFD
+                    ]
+                });
+                rcProcess.on('close', () => {
+                    fs.closeSync(outFD);
+                    fs.closeSync(errFD);
+                });
+            } catch (err) {
+                throw err;
+            }
+        }
+    }
+
+    console.log('Please wait for Hazelcast Remote Controller to start ...');
+
+    const retryCount = 10;
+
+    for (let i = 0; i < retryCount; i++) {
+        console.log('Trying to connect to Hazelcast Remote Controller (127.0.0.1:9701)...');
+        const addressReachable = await isAddressReachable('127.0.0.1', 9701, 5000);
+        if (addressReachable) {
+            return;
+        }
+        await new Promise(r => setTimeout(r, 1000));
+    }
+    throw `Could not reach to Hazelcast Remote Controller (127.0.0.1:9701) after trying ${retryCount} times.`;
+}
+const shutdownProcesses = () => {
+    console.log('Stopping remote controller and test processes...');
+    shutdownRC();
+    if (ON_WINDOWS) {
+        spawnSync('taskkill', ['/pid', testProcess.pid, '/f', '/t']); // simple sigkill not enough on windows
+    } else {
+        if (testProcess && testProcess.exitCode === null) testProcess.kill('SIGKILL');
+    }
+};
+const shutdownRC = () => {
+    if (ON_WINDOWS) {
+        spawnSync('taskkill', ['/pid', rcProcess.pid, '/f', '/t']);
+    } else {
+        if (rcProcess && rcProcess.exitCode === null) rcProcess.kill('SIGKILL');
+    }
+}
+
+if (process.argv.length === 3 || process.argv.length === 4) {
+    if (process.argv[2] === 'unit') {
+        if (process.argv.length === 4) {
+            testCommand = `node node_modules/mocha/bin/mocha --recursive -g ${process.argv[3]} "test/unit/**/*.js"`;
```

Comment:
Thought it would be good to add this functionality. We can run tests by regex using npm scripts like this:

`npm run test PNCounterBasicTest`

or run all tests as default behaviour as before
`npm run test`

Reply:
That's a great addition. Could you also describe this option in the following doc section: https://github.com/hazelcast/hazelcast-nodejs-client/blob/master/DOCUMENTATION.md#102-testing ?

Reply:
I described it in b670ec533e15b39044b66e9d681756b6e23e55b0, you can check